src/input.c

```diff
@@ -678,12 +848,51 @@ static void *input_thread_func(void *arg)
             }
         }
 
-        if (ch >= 0)
+        if (ch >= 0) {
             parse_char(input, (char) ch);
+        }
     }
     return NULL;
 }
 
+bool kitty_keyboard_protocol_supported(void)
+{
+    char buffer[128];
+    memset(buffer, 0, sizeof(buffer));
+
+    const char *query = "\033[?u";
+    if (write(STDOUT_FILENO, query, 4) != 4) {
+        perror("write");
+        return 1;
+    }
```

Comment:
Bug: Wrong return value on error.
It returns true (protocol supported) on failure.